torch/csrc/utils/tuple_parser.cpp

```diff
@@ -0,0 +1,85 @@
+#include "tuple_parser.h"
+
+#include "torch/csrc/DynamicTypes.h"
+#include "torch/csrc/autograd/python_variable.h"
+#include "python_numbers.h"
+
+namespace torch {
+
+TupleParser::TupleParser(PyObject* args, int num_args) : args(args), idx(0) {
+   int size = PyTuple_GET_SIZE(args);
+   if (num_args >= 0 && size != num_args) {
+     std::string msg("missing required arguments (expected ");
+     msg += std::to_string(num_args) + " got " + std::to_string(size) + ")";
+     throw std::runtime_error(msg);
+   }
+ }
+
+auto TupleParser::parse(bool& x) -> void {
```

Comment:
It would be nicer to be able to pass multiple arguments into the call, as you can do to `_appendData` in THD. The only downside is that it instantiates a new template for each configuration of arguments, but it's more readable.